src/components/users.js

```diff
@@ -472,7 +587,40 @@ export const UserEdit = props => {
             perPage={50}
             sort={{ field: "created_ts", order: "DESC" }}
           >
-            <Datagrid style={{ width: "100%" }}>
+            <Datagrid
+              style={{ width: "100%" }}
+              hasBulkActions={true}
+              header={<UserMediaDatagridHeader />}
+            >
+              <FunctionField
+                label="Image"
+                render={record => {
+                  let data = {
+                    title: record.upload_name,
+                    imgURL: `${localStorage.getItem(
+                      "base_url"
+                    )}/_matrix/media/v1/thumbnail/${localStorage.getItem(
+                      "home_server"
+                    )}/${record.media_id}?width=50&height=50&method=crop`,
+                    downloadURL: `${localStorage.getItem(
+                      "base_url"
+                    )}/_matrix/media/r0/download/${localStorage.getItem(
+                      "home_server"
+                    )}/${record.media_id}`,
+                  };
+                  if (record.media_type.startsWith("image")) {
+                    return (
+                      <ImageField
+                        record={data}
+                        source="imgURL"
+                        onClick={() => window.open(data.downloadURL)}
+                      />
+                    );
+                  } else {
+                    return "Preview unavailable";
+                  }
```

Comment:
Or something like this:

```suggestion
                  return (
                    <Avatar
                      src={data["imgURL"]}
                      onClick={() => window.open(data["downloadURL"])}
                      variant="square"
                    >
                      <BrokenImageIcon />
                    </Avatar>
                  );
```
With this fallback image:
```js
import BrokenImageIcon from "@material-ui/icons/BrokenImage";
```
Doc: https://mui.com/components/avatars/

In this case I would set `imgURL` to `null` if it is not an image. Otherwise `Avatar` would try to download / show the "not image" and creates a 404 request.
```js
                    imgURL: record.media_type.startsWith("image")
                      ? `${localStorage.getItem(
                          "base_url"
                        )}/_matrix/media/v1/thumbnail/${localStorage.getItem(
                          "home_server"
                        )}/${record.media_id}?width=50&height=50&method=crop`
                      : null,
```